rclcpp/include/rclcpp/parameter_client.hpp

```diff
@@ -185,6 +185,19 @@ class SyncParametersClient
     const std::string & remote_node_name = "",
     const rmw_qos_profile_t & qos_profile = rmw_qos_profile_parameters);
 
+  RCLCPP_PUBLIC
+  explicit SyncParametersClient(
+    rclcpp::Node * node,
```

Comment:
Instead a signature which took the component parts of the node that this client needs would be better, e.g. maybe the `NodeParameterInterface` and the `NodeBaseInterface` (and perhaps others?).

Reply:
You could keep a signature like this one which simply extracted those components from a `rclcpp::Node` and delegated them to another constructor. That way this could be more easily used with other forms of Node's like `rclcpp_lifecycle::LifecycleNode`.

Reply:
From what I have understood, SyncParameterClient only uses `NodeBaseInterface` for calling `spin_until_future_complete`.
All the other interfaces are used by AsyncParameterClient and are "extracted" from the node pointer in the AsyncParameterClient constructor. 
I do not get what's the advantage of having a SyncParameterClient constructor signature which takes interfaces, since all the logic using them is contained in the AsyncParameterClient